qutip/qip/compiler/gatecompiler.py

```diff
@@ -95,23 +126,138 @@ def decompose(self, gates):
+                elif len(tlist) == len(coeff):
+                    # continuos pulse
+                    step_size = tlist[1] - tlist[0]
+                    if compiled_coeffs[pulse_ind]:  # not first pulse
+                        temp_tlist = np.asarray(tlist)[1:]
+                        coeff = np.asarray(coeff)[1:]
+                    else:  # first pulse
+                        temp_tlist = np.asarray(tlist)[1:]
+                        coeff = np.asarray(coeff)
+                else:
+                    raise ValueError(
+                        "The shape of the compiled pulse is not correct.")
+                # If there is a idling time between the last pulse and
+                # the current one, we need to add zeros in between.
+                # We add sufficient number of zeros at the begining
+                # and the end of the idling to prevent wrong cubic spline.
+                if np.abs(start_time - compiled_tlist[pulse_ind][-1][-1]) \
+                        > step_size * 1.0e-6:
+                    idling_time = (
+                            start_time - compiled_tlist[pulse_ind][-1][-1]
+                        )
+                    if idling_time > 3 * step_size:
+                        idling_tlist1 = np.linspace(
+                            compiled_tlist[pulse_ind][-1][-1] + step_size/5,
+                            compiled_tlist[pulse_ind][-1][-1] + step_size,
+                            5
+                        )
+                        idling_tlist2 = np.linspace(
+                            start_time - step_size,
+                            start_time - step_size/5,
+                            5
+                        )
+                        idling_tlist = np.concatenate(
+                            [idling_tlist1, idling_tlist2])
+                    else:
+                        idling_tlist = np.arange(
+                            compiled_tlist[pulse_ind][-1][-1] + step_size,
+                            start_time - step_size, step_size
+                        )
+                    compiled_tlist[pulse_ind].append(idling_tlist)
+                    compiled_coeffs[pulse_ind].append(
+                        np.zeros(len(idling_tlist))
+                    )
+                    # The pulse should always start with 0
+                    # For rectangular pulse, len(tlist) = 0 and the above
+                    # idling_tlist is empty, hence this is necenssary.
+                    compiled_tlist[pulse_ind].append([start_time])
+                    compiled_coeffs[pulse_ind].append([0.])
+                # add the compiled pulse coeffs to the list
+                compiled_tlist[pulse_ind].append(temp_tlist + start_time)
+                compiled_coeffs[pulse_ind].append(coeff)
+
+        for i in range(num_pulses):
+            if not compiled_coeffs[i]:
+                compiled_tlist[i] = None
+                compiled_coeffs[i] = None
```

Comment:
I don't have a strong argument. I guess I made it like this because, in the end, they will be given to `QobjEvo`, which prefer `None` than `[]` for an empty `tlist`.